lib/puppet/ssl/certificate_authority.rb

```diff
@@ -356,10 +366,22 @@ def check_internal_signing_policies(hostname, csr, allow_dns_alt_names)
       raise CertificateSigningError.new(hostname), "CSR contains a public key that does not correspond to the signing key"
     end
 
+    auth_extensions = csr.request_extensions.select do |extension|
+      Puppet::SSL::Oids.subtree_of?('ppAuthCertExt', extension['oid'], true)
+    end
+
+    if auth_extensions.any? && !options[:allow_authorization_extensions]
+      ext_names = auth_extensions.map do |extension|
+        extension['oid']
+      end
+
+      raise CertificateSigningError.new(hostname), "CSR '#{csr.name}' contains authorization extensions (#{ext_names.join(', ')}), which are disallowed. Use `puppet cert --allow-authorization-extensions` to sign this request."
```

Comment:
Bike shed - instead of "disallowed" I would say "disallowed by default" - the first reading is a touch confusing. But again this is bike sheddy, periwinkle blue is also a lovely shade. :)